server/res/graph.html

```diff
@@ -13,62 +13,127 @@
+              for(key in datalist) {
+                dataset.push({
+                  y: datalist[key],
+                  type: 'scatter'
+                })
+              }
+              Plotly.plot(divId, dataset, layout, {displayModeBar: false});
+
+              setInterval(function(){
+                let index = 0;
+                for(key in datalist) {
+                  newdata = getData(key);
+                  console.log(newdata)
+                  if(newdata != '0' && key !== "flag") {
+                    Plotly.extendTraces(divId, {y: [[newdata]]}, [index]);
+                    //Plotly.update(divId, datalist[key], layout, index)
+                  }
+                  index++;
+                  counter++;
+                  // if(counter > 50) {
+                  //     Plotly.relayout(divId,{
+                  //       xaxis: {
+                  //           range: [counter-50,counter],
+                  //           fixedrange: true
+                  //       }
+                  //   });
+                  // }
+                    // if(counter % 100 == 0) {
+                    //     switch(currTrace) {
+                    //       case 1:
+                    //         Plotly.deleteTraces(divId, currTrace-1);
+                    //         currTrace = 2;
+                    //         Plotly.relayout(divId, {
+                    //           trace2: {
+                    //             visible: true
+                    //           },
+                    //           trace1: {
+                    //             visible: false
+                    //           }
+                    //         });
+                    //         break;
+                    //       case 2:
+                    //         Plotly.deleteTraces(divId, currTrace-1); //put the trace back in somehow
+                    //         currTrace = 1;
+                    //         Plotly.relayout(divId, {
+                    //           trace1: {
+                    //             visible: true
+                    //           },
+                    //           trace2: {
+                    //             visible: false
+                    //           }
+                             // })
+                        //}
+                    //}
+                  }
+              },60);
 
-            feedSocket2.onmessage = function(msg) {
-                last_data = parseInt(msg.data.split(",")[0]);
-                last_label = parseInt(msg.data.split(",")[1]);
+              // Do we assume that the configs always mach the number of data in the message?
+              websocket.onmessage = function(msg) {
+                console.log(msg)
+                msg_data = msg.data.split(",");
+                var index = 0;
+                for(i in configs) {
+                  let config = configs[i]
+                  if(index < msg_data.length) {
+                    datalist[config].push(msg_data[index]);
```

Comment:
You should do the remove of old data here. So when the size of the data > 500, start removing elements?